src/main/java/org/opensearch/neuralsearch/search/util/ScoreDocsMerger.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.util;
+
+import lombok.NoArgsConstructor;
+import org.apache.lucene.search.ScoreDoc;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.neuralsearch.search.util.HybridSearchResultFormatUtil.isHybridQueryScoreDocElement;
+
+/**
+ * Merges two ScoreDoc arrays into one
+ */
+@NoArgsConstructor
+public class ScoreDocsMerger<T extends ScoreDoc> {
```

Comment:
I think we should add name **HybridQuery** in the Name of the class as this operation are exclusive to the HybridQuery. like HybridQueryScoreDocsMerger

Reply:
ack, we use format that is specific for hybrid query results